src/sirocco/workgraph.py

```diff
@@ -317,6 +322,23 @@ def _set_shelljob_arguments(self, task: core.ShellTask):
         _, arguments = self.split_cmd_arg(task.resolve_ports(input_labels))
         workgraph_task_arguments.value = arguments
 
+    def _set_shelljob_filenames(self, task: core.ShellTask):
+        """set AiiDA ShellJob filenames for AvailableData entities"""
+
+        filenames = {}
+
+        for input_list in task.inputs.values():
```

Comment:
inputs can be a list of inputs because for shell task we can assign multiple inputs for one port and the list returns you all the inputs for this one port. I don't think we stress test this much so most of the time the input_list has just one element.
```suggestion
        for input_list in task.input_data_nodes:
```

Reply:
I think you need to handle all inputs separate as these can be potentially different files. Note that currently our code is also buggy in this regard. If we pass the outputs of a parametrized shell task to another task aiida will copy the file to the working directory. Since the output filename is the same for each parameter they will be overwritten. So only one file exists in the working directory of the next tasks that gets this as input. The solution is to give the outputs unique names.